pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -316,6 +317,7 @@ func (r *ReconcileClusterSync) Reconcile(request reconcile.Request) (reconcile.R
 			return reconcile.Result{}, err
 		}
 		// requeue immediately so that we reconcile soon after the ClusterSync is created
+		outcome = hivemetrics.ReconcileClusterSyncCreated
```

Comment:
i personally think this is magic and not really something some reading this code will immediately understand that the defer will use this outcome.

i would recommend we create a ReconcileObserver

```
robsrv := hivemetrics.NewOutcomeReconcileObserver(ControllerName.String(), logger) // this starts a timer and defautls the outcome to no-op
defer robsrv.ObserveControllerReconcileTime()

...
   // in some code somewhere.
    robsrv.SetOutcome(hivemetrics.ReconcileClusterSyncCreated)
```

Reply:
This looks marginally more clear, but what is the purpose of the timer you allude to?

Reply:
> This looks marginally more clear

to me the use if far more clearer, but :shrug: 

> but what is the purpose of the timer you allude to?

we have a observer struct to maintain the outcome, we could also just stop calling `start := time.Now()` at the top of each function and the creating new observer would call that already, less to work about.